### PR TITLE
Add IRulesRpcService 

### LINKED ISSUE #5162: Add IRulesRpcService

### PR DESCRIPTION
**Pull request by ugras-ergun-sonarsource**
Fixes #5162

### CODE REVIEW

File: src/SLCore/Service/Rules/IRulesRpcService.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        /// <summary>
+        /// Gets Rule Meta Data from SLCORE
+        /// </summary>
+        /// <param name="params"></param>
+        Task<object> GetEffectiveRuleDetailsAsync(GetEffectiveRuleDetailsParams @params);
+    }
+
+    public class GetEffectiveRuleDetailsParams
+    {
+        public string configurationScopeId;
```
Comment (georgii-borovinskikh-sonarsource) [inline src/SLCore/Service/Rules/IRulesRpcService.cs:37]: Convert to property

File: src/SLCore/Service/Rules/IRulesRpcService.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    public interface IRulesRpcService : ISLCoreService
+    {
+        /// <summary>
+        /// Gets Rule Meta Data from SLCORE
+        /// </summary>
+        /// <param name="params"></param>
+        Task<object> GetEffectiveRuleDetailsAsync(GetEffectiveRuleDetailsParams @params);
+    }
+
+    public class GetEffectiveRuleDetailsParams
```
Comment (georgii-borovinskikh-sonarsource) [inline src/SLCore/Service/Rules/IRulesRpcService.cs:35]: Suggestion: Move to separate file

File: src/SLCore/Service/Rules/IRulesRpcService.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+namespace SonarLint.VisualStudio.SLCore.Service.Rules
+{
+    public interface IRulesRpcService : ISLCoreService
+    {
+        /// <summary>
+        /// Gets Rule Meta Data from SLCORE
+        /// </summary>
+        /// <param name="params"></param>
+        Task<object> GetEffectiveRuleDetailsAsync(GetEffectiveRuleDetailsParams @params);
```
Comment (georgii-borovinskikh-sonarsource) [inline src/SLCore/Service/Rules/IRulesRpcService.cs:32]:
```suggestion
        Task<object> GetEffectiveRuleDetailsAsync(GetEffectiveRuleDetailsParams parameters);
```

File: src/SLCore/Service/Rules/IRulesRpcService.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.SLCore.Core;
+
+namespace SonarLint.VisualStudio.SLCore.Service.Rules
+{
+    public interface IRulesRpcService : ISLCoreService
```
Comment (georgii-borovinskikh-sonarsource) [inline src/SLCore/Service/Rules/IRulesRpcService.cs:26]: Missing JsonRpcClassAttribute with `rule` prefix value

**Review by georgii-borovinskikh-sonarsource — CHANGES_REQUESTED** [review]

File: src/SLCore/Service/Rules/IRulesRpcService.cs (hunk 2 above)
Comment (ugras-ergun-sonarsource) [reply src/SLCore/Service/Rules/IRulesRpcService.cs:35]: I think it makes sense to have these in the same file since they're tightly coupled.

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]

**Pull request merged**